modules/otel-troubleshoot-instrumentation.adoc

```diff
@@ -0,0 +1,91 @@
+
+To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
+
+[source,console]
+----
+oc get events -n <your-project>
+----
+
+You should see something like this:
+
+[source,console]
+----
+... Created container opentelemetry-auto-instrumentation
+... Started container opentelemetry-auto-instrumentation
+----
+
+If those messages are missing, then it means that was an issue while injecting the auto-instrumentation.
+
+. Inspect for the operator logs for instrumentation errors:
++
+[source,console]
+----
+oc logs -l app.kubernetes.io/name=opentelemetry-operator --container manager -n openshift-opentelemetry-operator --follow
+----
+
+. Confirm that the application pods you want to instrument are labeled correctly and the appropriate auto-instrumentation settings have been applied. For example, if you are instrumenting a Python application, you can check that the pods have the `instrumentation.opentelemetry.io/inject-python='true'` annotation:
++
+[source,console]
+----
+oc get pods -n <your-project> -o jsonpath='{range .items[?(@.metadata.annotations["instrumentation.opentelemetry.io/inject-python"]=="true")]}{.metadata.name}{"\n"}{end}'
+----
+
+You should double-check the annotation applied to the instrumentation object is the correct one for the language you are instrumenting. If there are multiple instrumentations in the same namespace, you should set its name in the annotation. Also, if the Instrumentation object is in a different namespace, you need to specify it in the annotation. You can find the accepted values in the link:otel-instrumentation.html["OpenTelemetry instrumentation configuration options"] section.
+
+When defining a Deployment, remember that annotations can be added in either {{spec.metadata.annotations}} or {{spec.template.metadata.annotations}}. Ensure that the auto-instrumentation annotation is added to {{spec.template.metadata.annotations}}.
+
+
+== Check the instrumentation libraries are generating telemetry data
+. Verify the endpoint that the instrumentation is transmitting data to:
++
+[source,console]
+----
+oc get instrumentation <instrumentation-name> -n <your-project> -o jsonpath='{.spec.endpoint}'
+----
++
+By default, the endpoint is `http://localhost:4317`. This will work if you deployed the OpenTelemetry Collector as a sidecar container in your application pod. If you are using a different endpoint, you can update it by editing the instrumentation object and redeploying your application.
+
+. Inspect your application logs for error messages that might suggest that the instrumentation is malfunctioning:
++
+[source,console]
+----
+oc logs <application-pod> -n <your-project>
+----
++
+Some common errors include:
+* Unable to automatically infer a package for instrumentation
+* Unsupported application framework or version
```

Comment:
Can I delete these three lines?
My reason is that whatever error message the user will get is what the user will have to deal with.
WDYT?

Reply:
I'm fine with that.

Reply:
Done.